src/mgds/pipelineModules/DropTags.py

```diff
@@ -0,0 +1,155 @@
+            with open(sptags) as special_tags_file:
+                return [line.rstrip('\n') for line in special_tags_file]
+        elif (sptags.endswith(".csv") and os.path.isfile(sptags)):
+            with open(sptags, 'r') as special_tags_file:
+                splist = []
+                for row in csv.reader(special_tags_file):
+                    splist.append(row[0])
+                return splist
+        else:
+            return [tag.strip() for tag in sptags.split(delim)]
+
+    #parse regex expressions, create new special list based on matches   
+    def parse_regex(self, splist_in, taglist):
+        splist_out = []
+        regex_spchars = set(".^$*+?!\{\}\[\]|()\\")
+        for c in splist_in:
+            if any((a in regex_spchars) for a in c):    #only do regex matching if tag contains special character
+                c = c.replace("\)", "\\\\\)")
+                c = c.replace("\(", "\\\\\(")   #hopefully fix issues caused by "\(\)" syntax without affecting other regex
+                r = re.compile(c)
+                for s in taglist:
+                    if r.fullmatch(s):
+                        splist_out.append(s)
+            else:
+                splist_out.append(c)
+        return splist_out    
+    
+    #change probability evaluated against random() depending on mode
+    def probability_weighted(self, p, mode, i, len):
+        if mode == "RANDOM":
+            return float(p)
+        elif mode == "RANDOM WEIGHTED":
+            return float(p*(i/len))
+        else:   #catch errors
+            return float(p)
+    
+    def get_item(self, variation: int, index: int, requested_name: str = None) -> dict:
+        text = self._get_previous_item(variation, self.text_in_name, index)
+        delimiter = self._get_previous_item(variation, self.delimiter_in_name, index)
+        keep_tags_count = self._get_previous_item(variation, self.keep_tags_count_in_name, index)
+        enabled = self._get_previous_item(variation, self.enabled_in_name, index)
+        probability = self._get_previous_item(variation, self.probability_in_name, index)
+        dropout_mode = self._get_previous_item(variation, self.dropout_mode_in_name, index)
+        special_tag_mode = self._get_previous_item(variation, self.special_tag_mode_in_name, index)
+        special_tags = self._get_previous_item(variation, self.special_tags_in_name, index)
+        regex_enabled = self._get_previous_item(variation, self.regex_enabled_in_name, index)
+        rand = self._get_rand(variation, index)
+
+        if enabled and (probability > 0):
+            #convert inputs to lists and set up final output list (pruned) and white/blacklist (special)
+            tags = [tag.strip() for tag in text.split(delimiter)]
+            keep_tags = tags[:keep_tags_count]
+            dropout_tags = tags[keep_tags_count:]
+            pruned_tags = []
+            special_tags_prelist = []
+            special_tags_list = []
```

Comment:
Thanks, I had them in there from when the file loading and regex sections were not separate functions, but can probably be removed now. I'll have to make sure there aren't any cases where those functions could be skipped and the variables never defined and cause errors later on, if so I think it would be beneficial to keep them included